app/controllers/api/api_controller.rb

```diff
@@ -26,7 +26,7 @@ def entrypoint_versions
       ApiConfig.version.definitions.select(&:ident).collect do |version_specification|
         {
           :name => version_specification[:name],
-          :href => "#{@req.api_prefix}/#{version_specification[:ident]}"
+          :href => @req.api_prefix.include?(version_specification[:ident]) ? @req.api_prefix : "#{@req.api_prefix}/#{version_specification[:ident]}"
```

Comment:
can you double check the code in parser/request_adapter.rb ?  I thought api_prefix always includes the version so we may not want to use it.

issue remaining, I've added 2.4.0 as a version def in the api.yml and got this got fetching GET /api/v2.4.0

```
  "version": "3.0.0-pre",
  "versions": [
    {
      "name": "3.0.0-pre",
      "href": "http://localhost:3000/api/v2.4.0/v3.0.0-pre"
    },
    {
      "name": "2.4.0",
      "href": "http://localhost:3000/api/v2.4.0"
    }
  ],
```